tracer/src/Datadog.Trace/HttpOverStreams/HttpHeaderHelperBase.cs

```diff
@@ -16,8 +16,9 @@ internal abstract class HttpHeaderHelperBase
 
         public Task WriteLeadingHeaders(HttpRequest request, TextWriter writer)
         {
-            var leadingHeaders =
-                $"{request.Verb} {request.Path} HTTP/1.1{DatadogHttpValues.CrLf}Host: {request.Host}{DatadogHttpValues.CrLf}Accept-Encoding: identity{DatadogHttpValues.CrLf}Content-Length: {request.Content?.Length ?? 0}{DatadogHttpValues.CrLf}{MetadataHeaders}";
+            var leadingHeaders = request.Content?.Length is { } contentLength
+                                     ? $"{request.Verb} {request.Path} HTTP/1.1{DatadogHttpValues.CrLf}Host: {request.Host}{DatadogHttpValues.CrLf}Accept-Encoding: identity{DatadogHttpValues.CrLf}Content-Length: {contentLength}{DatadogHttpValues.CrLf}{MetadataHeaders}"
+                                     : $"{request.Verb} {request.Path} HTTP/1.1{DatadogHttpValues.CrLf}Host: {request.Host}{DatadogHttpValues.CrLf}Accept-Encoding: identity{DatadogHttpValues.CrLf}{MetadataHeaders}";
```

Comment:
I think you could have extracted the `Content-Length: {contentLength}` bit into its own variable, and set it to empty when content-length is null. Here it will be hard to keep the two sets of headers in sync.

Reply:
Just to save iteration/CI time I'm going to make that fix in https://github.com/DataDog/dd-trace-dotnet/pull/5446